app/routes/ui/scenario.py

```diff
@@ -0,0 +1,45 @@
+from library.postgres import (
+    get_scenario_by_friendly_name_from_postgres,
+    get_scene_from_postgres,
+)
+from models.scenario import Scenario
+from routes.base import BaseUIHandler
+
+
+class UIScenarioHandler(BaseUIHandler):
+    """
+    Handle routes that have {scenario_friendly_name}/{scene_number}
+    """
+
+    async def get(self, scenario_friendly_name, scene_number=0):
+        try:
+            scene_number_int = int(scene_number)
+            # TODO: caching will be dealt with in a later PR
+            scenario_obj: Scenario = await get_scenario_by_friendly_name_from_postgres(
+                scenario_friendly_name, self.db_session
+            )
+            if not scenario_obj.is_published:
+                raise ValueError("This Scenario is currently not accessible")
+
+            if scene_number_int >= len(scenario_obj.scene_ids) or scene_number_int < 0:
+                raise ValueError("Invalid Scene")
```

Comment:
Order of checks? Should we check invalid before is published?

Pros: if they see it's not accessible, they'll assume it exists even though it might not.

Cons: malicious users can go through and find which scene ids exist, published or not.

I'll let you decide which is best - won't block the PR for this.

Reply:
No, so this order is what the most secure way. If a scenario is not published, then we should leak *no* information, hence why the scene ids check is only after ensuring that the scenario is supposed to be accessible.